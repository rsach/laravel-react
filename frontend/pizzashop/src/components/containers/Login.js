import { connect } from 'react-redux'
import { login} from '../../actions'


import Login from "../ui/Login";

const mapStateToProps = (state,props) =>
    ({

    });

const mapDispatchToProps = dispatch =>
    ({
        onLogin(body){
            dispatch(
                login(body)
            )
        }

    });

export default connect(mapStateToProps,mapDispatchToProps)(Login)
