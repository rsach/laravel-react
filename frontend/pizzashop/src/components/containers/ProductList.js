import { connect } from 'react-redux'
import {addToCart, fetchProducts} from '../../actions'
import ProductList from "../ui/ProductList";

const mapStateToProps = (state,props) =>
    ({
        products:state.products.products,
    });

const mapDispatchToProps = dispatch =>
    ({
        onAddToCart(product){
            dispatch(
                addToCart(product)
            )
        },
        fetchProducts() {
            dispatch(
                fetchProducts('')
            )
        }
    });

export default connect(mapStateToProps,mapDispatchToProps)(ProductList)
