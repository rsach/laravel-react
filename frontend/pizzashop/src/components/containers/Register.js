import { connect } from 'react-redux'
import { register} from '../../actions'


import Register from "../ui/Register";

const mapStateToProps = (state,props) =>
    ({

    });

const mapDispatchToProps = dispatch =>
    ({
        onSignUp(body){
            console.log(body)
            dispatch(
                register(body)
            )

            return body
        }

    });

export default connect(mapStateToProps,mapDispatchToProps)(Register)
