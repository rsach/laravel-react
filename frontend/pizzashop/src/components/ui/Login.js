import React, {Component} from 'react';
import PropTypes from "prop-types";
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
// import AppBar from 'material-ui/AppBar';
import AppBar from '@material-ui/core/AppBar';



import history from "../../history";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";

class Login extends Component {
    constructor(props) {
        super(props);

        // reset login status


        this.state = {
            email: '',
            password: '',
            submitted: false
        };


        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleRedirect = this.handleRedirect.bind(this);
    }




    handleSubmit(e) {
        e.preventDefault();

        this.setState({ submitted: true });
        const { email, password } = this.state;
        const {onLogin=f=>f} = this.props;

        if (email && password) {

            onLogin({email, password});
        }
    }

    handleRedirect(e) {
        e.preventDefault();
        history.push('/register')
    }

    render() {


        return (

            <div>
                <MuiThemeProvider>
                    <div>
                        <AppBar
                            title="Login"
                        />
                        <div className="flex  flex-center">
                            <div className="form flex flex-column flex-center">


                                <TextField
                                    id="email"
                                    data-testid="email"
                                    variant="outlined"
                                    placeholder="Enter your Email"
                                    label="Email"
                                    onChange = {(event) => this.setState({email:event.target.value})}
                                />
                                <br/>
                                <TextField
                                    data-testid="password"

                                    id="password"
                                    variant="outlined"
                                    type="password"
                                    placeholder="Enter your Password"
                                    label="Password"
                                    onChange = {(event) => this.setState({password:event.target.value})}
                                />
                                <br/>
                                <div className="flex button-margin flex-space-between">
                                    <Button                    id="login"                 data-testid="login"
                                                                                variant="contained" color="primary"  onClick={(event) => this.handleSubmit(event)}>Login</Button>

                                        <Button id="register" variant="contained"        data-testid="register"
                                                onClick={(e) => this.handleRedirect(e)}  color="secondary"    >
                                            Register
                                        </Button>




                                </div>
                            </div>

                        </div>

                    </div>
                </MuiThemeProvider>
            </div>
        );
    }
}


Login.propTypes = {
    onLogin: PropTypes.func.isRequired
};

export default Login

