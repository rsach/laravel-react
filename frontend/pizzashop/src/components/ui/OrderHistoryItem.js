
import React from 'react'

import '../../stylesheets/Menu.scss'
import ExpansionPanel from "@material-ui/core/ExpansionPanel";
import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import Typography from "@material-ui/core/Typography";
import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails";
import moment from 'moment';
import CartItem from "./cart-item";



const OrderHistoryItem = ({item}) => {


    let total = 0;
    ((item || {}).order_items || []).forEach(res => total += (+res.price * +res.quantity));

    const order_on = moment((item || {}).created_at ).format('MM-DD-YYYY hh:mm');
    return (<ExpansionPanel>
        <ExpansionPanelSummary
            expandIcon={<ExpandMoreIcon/>}
            aria-controls="panel1a-content"
            id="panel1a-header"
        >

                <div className="flex past-order-container flex-space-between">
                    <Typography id="total"> Total: {(item || {}).currency === 'dollar' ? '$' : 'â‚¬' } {total}</Typography>

                    <Typography>
                        Ordered On: {order_on}
                    </Typography>
                </div>

        </ExpansionPanelSummary>
        <ExpansionPanelDetails>


            <div className="product-list flex flex-column">


                {
                    ((item || {}).order_items || []).map(res => ({...res, ...res.menu}))
                                    .map((res,i) =>
                                                <CartItem key={i} currency={item.currency} item={res} />
                    )
                }




            </div>


        </ExpansionPanelDetails>
    </ExpansionPanel>
    );

};

OrderHistoryItem.propTypes = {
    item: (props) => (!(typeof props.item === 'object')) ?
        new Error('Order must be an object') :
        null,
};

export default OrderHistoryItem