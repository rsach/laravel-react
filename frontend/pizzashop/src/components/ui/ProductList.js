
import '../../stylesheets/index.scss'
import Product from "../containers/Product";
import React from 'react'
import AppBar from "material-ui/AppBar";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";

const ProductList = ({products = []}) =>
    (
                <div>
                    <MuiThemeProvider>
                        <div>
                            <AppBar
                               className="fixed" title="Pizza List"
                            />

                            <div className="product-list flex flex-wrap">

                                {
                                    (products || []).map((product,i) => <Product key={i} product={product} />)
                                }

                            </div>

                        </div>
                    </MuiThemeProvider>
                </div>

            );




 

ProductList.propTypes = {
    products: (props) => (!Array.isArray(props.products)) ?
        new Error("Product List property must be an array") :
            null,


};

export default ProductList