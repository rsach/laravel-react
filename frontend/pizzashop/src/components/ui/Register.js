import React, {Component} from 'react';

import PropTypes from "prop-types";


import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';


import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import history from "../../history";


class Register extends Component {
    constructor(props) {
        super(props);

        this.state = {
            user: {
                name: '',
                email: '',
                password: ''
            },
            submitted: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleRedirect = this.handleRedirect.bind(this);
    }

    handleChange(newValue) {
        // const { name, value } = event;
        const { user } = this.state;
        this.setState({
            user: {
                ...user,
                ...newValue
            }
        });
    }

    handleSubmit(event) {
        event.preventDefault();

        this.setState({ submitted: true });
        const { user } = this.state;
        const {onSignUp=f=>f} = this.props;

        if (user.name && user.email && user.password) {

            onSignUp(user);
        }
    }

    handleRedirect(e) {
        e.preventDefault();
        history.push('/login')
    }

    render() {

        // const { user, submitted } = this.state;
        return (
            <div>
                <MuiThemeProvider>
                <div>
                    <AppBar
                        title="Register"
                    />
                    <div className="flex  flex-center">
                        <div className="form flex flex-column">

                                <TextField
                                    id="name"
                                    variant="outlined"

                                placeholder="Enter your Name"
                                label="Name"
                                onChange = {(event) => this.handleChange({name: event.target.value})}
                            />


                            <br/>
                            <TextField
                                id="email"
                                placeholder="Enter your Email"
                                type="email"
                                variant="outlined"

                                label="Email"
                                onChange = {(event) => this.handleChange({email: event.target.value})}
                            />
                            <br/>
                            <TextField
                                id="password"
                                type = "password"
                                variant="outlined"

                                placeholder="Enter your Password"
                                label="Password"
                                onChange = {(event) => this.handleChange({password: event.target.value})}
                            />
                            <br/>
                            <div className="flex button-margin flex-space-between">
                                <Button id="register" label="Register" variant="contained"  color="primary"  onClick={(event) => this.handleSubmit(event)}> Register </Button>


                                    <Button id="login" variant="contained" label="Login" color="secondary" onClick={(e) => this.handleRedirect(e)}    >

                                        Login
                                    </Button>



                            </div>
                        </div>
                    </div>
                </div>
            </MuiThemeProvider>
            </div>
        );
    }
}


Register.propTypes = {
    onSignUp: PropTypes.func,
};

export default Register
