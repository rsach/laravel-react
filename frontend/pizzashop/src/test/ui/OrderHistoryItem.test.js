import React from 'react';
import { render } from '@testing-library/react';
import Login from "../../components/ui/Login";
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Register from "../../components/ui/Register";
import SuccessOrder from "../../components/ui/SuccessOrderPage";
import OrderHistoryItem from "../../components/ui/OrderHistoryItem";
import ProductList from "../../components/ui/ProductList";

Enzyme.configure({ adapter: new Adapter() });


jest.mock('moment', () => () => ({format: () => '2018–01–30T12:34:56+00:00'}));
jest.mock('moment', () => () => ({format: () => '2018–01–30T12:34:56+00:00'}));

describe('<OrderHistoryItem/> spec', () => {


    function setup(prop) {
        const props = {
            item: {

                order_items: [],
                created_at: new Date(),
                ...prop
            }
        };
        const wrapper = shallow(<OrderHistoryItem {...props} />);
        return {
            props,  wrapper
        }
    }



    it('renders the component', () => {

        const {wrapper} = setup({order_items: null})

        expect(wrapper.firstChild).toMatchSnapshot();
    });

    it('renders the component', () => {

        const {wrapper} = setup({order_items: [1, 2]});

        expect(wrapper.find('CartItem').length).toBe(2);
    });

    it('renders the component', () => {
        const spy = jest.spyOn(console, 'error');

        const wrapper = shallow(<OrderHistoryItem />);

        expect(spy).toHaveBeenCalledTimes(1);
    });

    it('renders the component', () => {
        const prop = {
            id:1,
            user_id:1,
            currency:"euro",
            created_at:"2019-12-29 14:29:44",
            updated_at:"2019-12-29 14:29:44",
            orders_items: [{
                id:1,
                menu_id:2,
                quantity:2,
                price:"7.00",
                created_at:"2019-12-29 14:29:44",
                updated_at:"2019-12-29 14:29:44",
                order_id:1,
                menu: {
                    id:2,
                    name:"Prof. Sylvan Mertz MD",
                    price:7,
                    picture:"https://lorempixel.com/640/480/?93343",
                    created_at:"2019-12-29 14:27:07",
                    updated_at:"2019-12-29 14:27:07",
                }


            }]
        }

        const {wrapper} = setup({...prop});

        console.log(wrapper.find('#total').text())
        expect(wrapper.find('#total').text().includes('€')).toBeTruthy();
        // expect(wrapper.find('#total').text().includes('14')).toBeTruthy();
    });


    it('renders the component', () => {
        const prop = {
            id:1,
            user_id:1,
            currency:"dollar",
            created_at:"2019-12-29 14:29:44",
            updated_at:"2019-12-29 14:29:44",
            orders_items: [{
                id:1,
                menu_id:2,
                quantity:2,
                price:"7.00",
                created_at:"2019-12-29 14:29:44",
                updated_at:"2019-12-29 14:29:44",
                order_id:1,
                menu: {
                    id:2,
                    name:"Prof. Sylvan Mertz MD",
                    price:7,
                    picture:"https://lorempixel.com/640/480/?93343",
                    created_at:"2019-12-29 14:27:07",
                    updated_at:"2019-12-29 14:27:07",
                }


            }]
        }

        const {wrapper} = setup({...prop});

        console.log(wrapper.find('#total').text())
        expect(wrapper.find('#total').text().includes('$')).toBeTruthy();
        // expect(wrapper.find('#total').text().includes('14')).toBeTruthy();
    });

});