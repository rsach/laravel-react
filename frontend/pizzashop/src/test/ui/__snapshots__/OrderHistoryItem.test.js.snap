// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderHistoryItem/> spec renders the component 1`] = `undefined`;
